src/analysis/lattice.h

```diff
@@ -23,35 +23,33 @@ inline LatticeComparison reverseComparison(LatticeComparison comparison) {
   }
 }
 
-template<typename Lattice>
-constexpr bool has_getBottom =
-  std::is_invocable_r<typename Lattice::Element,
-                      decltype(&Lattice::getBottom),
-                      Lattice>::value;
-template<typename Lattice>
-constexpr bool has_compare =
-  std::is_invocable_r<LatticeComparison,
-                      decltype(&Lattice::compare),
-                      Lattice,
-                      const typename Lattice::Element&,
-                      const typename Lattice::Element&>::value;
-template<typename Element>
-constexpr bool has_makeLeastUpperBound =
-  std::is_invocable_r<void,
-                      decltype(&Element::makeLeastUpperBound),
-                      Element,
-                      const Element&>::value;
-template<typename Element>
-constexpr bool has_isTop =
-  std::is_invocable_r<bool, decltype(&Element::isTop), const Element>::value;
```

Comment:
What happened to `isTop` in C++20?

Reply:
I don't think we need it for anything, so I didn't bother adding it back into the requirements.